src/demo.re

```diff
@@ -0,0 +1,66 @@
+/* 
+   Goal: 
+     conjugate('estudiar', { tense: 'present', subject: 'ella' });
+       should return estudia
+ */
+
+type subject = 
+| Yo
+| Tu
+| Ella
+| Nosotras
+| Vosotras
+| Ellas;
+
+type ending = 
+| Ar
+| Er
+| Ir;
+
+let getSubjectVariant subject => {
+  switch subject {
+    | "yo" => Yo
+    | "tu" => Tu
+    | "ella" => Ella
+    | "nosotras" => Nosotras
+    | "vosotras" => Vosotras
+    | _ => Ellas
+  }
+};
+
+let getEndingVariant ending => {
+  switch ending {
+    | "ar" => Ar
+    | "er" => Er
+    | _ => Ir
+  }
+};
+
+let genPresentEnding ending subject => {
+  Js.log("subject", subject);
+  switch subject {
```

Comment:
So one of the really cool things about Reason pattern matching is you can match on multiple things at the same time. E.g.,

```reason
switch (subject, ending) {
| (Yo, _) => "yo"
| (Tu, Ar) => "as"
| (Tu, _) => "es"
| (Ella, Ar) => "a"
| (Ella, _) => "e"
| (Nosotras, _) => "amos"
| (Vosotras, _) => "\195\161is" /* Reason source code is not Unicode, btw */
| (Ellas, _) => "an"
};
```
This keeps your matches really nice and flat.